src/sonic-config-engine/minigraph.py

```diff
@@ -991,6 +995,8 @@ def parse_linkmeta(meta, hname):
                 lower_tor_hostname = value
             elif name == "AutoNegotiation":
                 auto_negotiation = value
+            elif name == "MacSecEnabled":
+                macsec_enabled = value
```

Comment:
>macsec_enabled [](http://example.com/codeflow?start=16&length=14)

An explicit `False` value will override if `macsec_enabled = True` in previous iteration. I think this is not expected.

Reply:
My understanding is as below for each interface there is a LinkMeta . It is initialized False and looped per property for that port. 

```
            <a:LinkMetadata>
                <a:Name i:nil="true"/>
                <a:Properties>
                    <a:DeviceProperty>
                        <a:Name>MacSecEnabled</a:Name>
                        <a:Reference i:nil="true"/>
                        <a:Value>True</a:Value>
                    </a:DeviceProperty>
                </a:Properties>
                <a:Key>STG01-0101-0400-01T2-lc3:Eth48-ASIC0;STG30-0100-0100-02RHE:Ethernet4/7/1</a:Key>
            </a:LinkMetadata>
            <a:LinkMetadata>
                <a:Name i:nil="true"/>
                <a:Properties>
                    <a:DeviceProperty>
                        <a:Name>MacSecEnabled</a:Name>
                        <a:Reference i:nil="true"/>
                        <a:Value>True</a:Value>
                    </a:DeviceProperty>
                </a:Properties>
                <a:Key>STG01-0101-0400-01T2-lc3:Eth56-ASIC0;STG30-0100-0100-02RHE:Ethernet4/8/1</a:Key>
            </a:LinkMetadata>
```